utils.js

```diff
@@ -54,13 +48,11 @@ exports.normalizeScopeSets = function(scopesets) {
  * Also expressed as ('a' and 'b') or 'c'.
  */
 exports.scopeMatch = function(scopePatterns, scopesets) {
-  var scopesets = exports.normalizeScopeSets(scopesets);
+  exports.validateScopeSets(scopesets);
   if (typeof(scopePatterns) == 'string') {
     scopePatterns = [scopePatterns];
   }
```

Comment:
Or do we want to validate everything... Ahh, maybe that makes sense.. /me thinking...